### PR TITLE
Lots of code cleanup

### PR DESCRIPTION
**Pull request by Azami7**
* Made a single instance of Ollivanders2Common that everything can use
* Moved player-related common spells to new O2PlayerCommon class
* Made common functions not static since it was getting hard to remember which were static and which weren't and nearly every class needs some of the non-static ones so an object has to exist anyway
* Removed import *'s
* Fixed incomplete javadoc
* Moved OEffect in to Effect package
* fixed varaible and method access permissions
* Moved ExtraDimensional in to StationarySpells since it is a child class
* Added plugin callback to StationarySpellObj
* Moved player classes in to new Player package
* Added new class O2PlayerCommon for common player-specific data and functions
* Clean up of warnings, etc on player classes
* Changed default book author to "Unknown"

### CODE REVIEW

File: Ollivanders/src/net/pottercraft/Ollivanders2/Book/Book.java [hunk 1]
```diff
@@ -95,19 +96,17 @@ private void writeSpellBookMeta (ItemStack bookItem)
...
-      {
-         bookContents.add(potion);
-      }
+
+      bookContents.addAll(potions);
 
       for (String content : bookContents)
       {
-         String name = BookTexts.getName(content);
+         String name = p.books.spellText.getName(content);
```
Comment (tcubansk) [inline Ollivanders/src/net/pottercraft/Ollivanders2/Book/Book.java]: Where is does p get set to something besides null?

File: Ollivanders/src/net/pottercraft/Ollivanders2/Prophecy.java [hunk 2]
```diff
@@ -86,39 +82,35 @@ public PotionEffect toPotionEffect ()
...
+      int durs = (dur / (second));
       lore.add(durh + " hours, " + durm + " minutes, and " + durs + " seconds");
       return lore;
    }
 
-   public boolean isActive ()
+   boolean isActive ()
    {
       long now = System.currentTimeMillis();
       if (now > this.start && now < this.start + this.duration)
```
Comment (tcubansk) [inline Ollivanders/src/net/pottercraft/Ollivanders2/Prophecy.java]: return (now > this.start && now < this.start + this.duration);

Several methods in this class could have less lines

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/ExtraDimensional.java [hunk 3]
```diff
@@ -66,11 +64,11 @@ private void createSpace ()
...
    }
 
-   public Set<Block> getBlocksInCube (Location loc)
+   private Set<Block> getBlocksInCube (Location loc)
    {
       Block center = loc.getBlock();
-      int blockRadius = (int) (dimenRadius + 1);
-      Set<Block> blockList = new HashSet<Block>();
+      int blockRadius = dimenRadius + 1;
+      Set<Block> blockList = new HashSet<>();
```
Comment (tcubansk) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/ExtraDimensional.java]: nitpick: blockList is a Set

Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/ExtraDimensional.java]: Yea, I am not sure if this code works or not so I didn't want to refactor it too far until I know how it works.

File: Ollivanders/src/net/pottercraft/Ollivanders2/Prophecy.java (hunk 2 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/Prophecy.java]: Prophecy is totally broken so I didn't send time cleaning this one up much.

File: Ollivanders/src/net/pottercraft/Ollivanders2/Book/Book.java (hunk 1 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/Book/Book.java]: Oh good catch, it needs to be passed in to the constructor for Books

Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/Book/Book.java]: Added it to the constructor for Book

**Comment by Azami7** [comment]
Merged via command line.

**Pull request closed without merging**